Java docs generation error for janusgraph-driver during a build process.

Java docs error is received for `janusgraph-driver` during a build process. The folder `janusgraph-driver/target/apidocs` is empty after the build is successfully finished.

Expected behaviour:
Java docs are generated for `janusgraph-driver` module and populated in corresponding `apidocs` folder after the build is finished.

For confirmed bugs, please report:
- Version: 0.5.0-SNAPSHOT
- Steps to Reproduce: `mvn clean install -DskipTests=true`

### Stack Trace (if you have one)
```
...
[INFO] --- maven-javadoc-plugin:2.10.4:jar (attach-javadocs) @ janusgraph-core ---
[INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:javadoc' has not been previously called for the module: 'org.janusgraph:janusgraph-driver:jar:0.5.0-SNAPSHOT'. Trying to invoke it...
[WARNING] Creating fake javadoc directory to prevent repeated invocations: /home/ws/janusgraph/janusgraph-driver/target/apidocs
[ERROR] Error fetching link: /home/ws/janusgraph/janusgraph-driver/target/apidocs/package-list. Ignored it.
[INFO] 
100 warnings
[WARNING] Javadoc Warnings
...
```

--- a/janusgraph-all/pom.xml
+++ b/janusgraph-all/pom.xml
@@ -25,6 +25,11 @@
     <dependencies>
         <dependency>
             <groupId>org.janusgraph</groupId>
+            <artifactId>janusgraph-driver</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.janusgraph</groupId>
             <artifactId>janusgraph-core</artifactId>
             <version>${project.version}</version>
         </dependency>

--- a/janusgraph-core/pom.xml
+++ b/janusgraph-core/pom.xml
@@ -23,16 +23,6 @@
         <!-- TinkerPop -->
         <dependency>
             <groupId>org.apache.tinkerpop</groupId>
-            <artifactId>gremlin-core</artifactId>
-            <exclusions>
-                <exclusion>
-                    <artifactId>gossip</artifactId>
-                    <groupId>org.sonatype.gossip</groupId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.tinkerpop</groupId>
             <artifactId>gremlin-groovy</artifactId>
         </dependency>
         <dependency>
@@ -118,24 +108,6 @@
         </testResources>
         <plugins>
             <plugin>
-                <groupId>org.codehaus.gmavenplus</groupId>
-                <artifactId>gmavenplus-plugin</artifactId>
-                <version>${gmavenplus.version}</version>
-                <executions>
-                    <execution>
-                        <goals>
-                            <goal>generateStubs</goal>
-                            <goal>compile</goal>
-                            <goal>testGenerateStubs</goal>
-                            <goal>testCompile</goal>
-                        </goals>
-                        <configuration>
-                            <providerSelection>1.8</providerSelection>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-            <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
                 <configuration>
@@ -145,24 +117,6 @@
                         </manifestEntries>
                     </archive>
                 </configuration>
-            </plugin>
-            <plugin>
-                <artifactId>maven-javadoc-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>attach-javadocs</id>
-                        <phase>package</phase>
-                        <goals>
-                            <goal>jar</goal>
-                        </goals>
-                        <configuration>
-                            <windowtitle>JanusGraph ${project.version} API</windowtitle>
-                            <doctitle>JanusGraph ${project.version} API</doctitle>
-                            <!-- Quiet mode still shows warnings and errors -->
-                            <quiet>true</quiet>
-                        </configuration>
-                    </execution>
-                </executions>
             </plugin>
             <plugin>
                 <artifactId>maven-shade-plugin</artifactId>
@@ -218,36 +172,6 @@
                     </execution>
                 </executions>
             </plugin>
-
         </plugins>
     </build>
-
-    <profiles>
-        <!-- This profile forces gpg-plugin to run after shade-plugin.
-             Otherwise, Maven will sign the unshaded jar, then shade the jar
-             (giving it completely different contents and breaking the sig) -->
-        <profile>
-            <id>janusgraph-release</id>
-            <build>
-                <plugins>
-                    <plugin>
-                        <artifactId>maven-shade-plugin</artifactId>
-                        <executions>
-                            <execution>
-                                <id>shade-gremlin-groovy</id>
-                            </execution>
-                        </executions>
-                    </plugin>
-                    <plugin>
-                        <artifactId>maven-gpg-plugin</artifactId>
-                        <executions>
-                            <execution>
-                                <id>sign-artifacts</id>
-                            </execution>
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
 </project>

--- a/janusgraph-driver/pom.xml
+++ b/janusgraph-driver/pom.xml
@@ -19,6 +19,12 @@
         <dependency>
             <groupId>org.apache.tinkerpop</groupId>
             <artifactId>gremlin-core</artifactId>
+            <exclusions>
+                <exclusion>
+                    <artifactId>gossip</artifactId>
+                    <groupId>org.sonatype.gossip</groupId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.apache.tinkerpop</groupId>
@@ -97,11 +103,67 @@
                     <plugin>
                         <artifactId>maven-jar-plugin</artifactId>
                     </plugin>
+                    <!-- Forces gpg-plugin to run after shade-plugin.
+                     Otherwise, Maven will sign the unshaded jar, then shade the jar
+                     (giving it completely different contents and breaking the sig) -->
+                    <plugin>
+                        <artifactId>maven-shade-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>shade-gremlin-groovy</id>
+                            </execution>
+                        </executions>
+                    </plugin>
                     <plugin>
                         <artifactId>maven-gpg-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>sign-artifacts</id>
+                            </execution>
+                        </executions>
                     </plugin>
                 </plugins>
             </build>
         </profile>
     </profiles>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.gmavenplus</groupId>
+                <artifactId>gmavenplus-plugin</artifactId>
+                <version>${gmavenplus.version}</version>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>generateStubs</goal>
+                            <goal>compile</goal>
+                            <goal>testGenerateStubs</goal>
+                            <goal>testCompile</goal>
+                        </goals>
+                        <configuration>
+                            <providerSelection>1.8</providerSelection>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>attach-javadocs</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                        <configuration>
+                            <windowtitle>JanusGraph ${project.version} API</windowtitle>
+                            <doctitle>JanusGraph ${project.version} API</doctitle>
+                            <!-- Quiet mode still shows warnings and errors -->
+                            <quiet>true</quiet>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
 </project>

